packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -36,30 +37,41 @@ contract TaikoL2 is EssentialContract, TaikoL2Signer, ICrossChainSync {
 
     // A hash to check the integrity of public inputs.
     bytes32 public publicInputHash; // slot 3
-    // Will init it with value 1 in genesis generation script to simply enable
-    // the L2 1559.
-    uint128 public gasExcess; // slot 4
-    uint64 public latestSyncedL1Height; // slot 5
+    uint64 public latestSyncedL1Height; // slot 4
```

Comment:
@adaki2004 somehow I prefer this setter to be added to a subcontract of TaikoL2.sol and keep TaikoL2.sol as is. Do you think it's a better idea?

Reply:
You mean this one:
`setConfig()` ?

TBH i dont know if it makes sense to create a contract just for 3 variables.. (?).

Reply:
I mean add a TaikoL2EIP1559Configurable.sol so inherit  TaikoL2 then add/override whatever functions necessary.

Reply:
What are you doing here ? I totally do not get it why we need to work like this.
https://github.com/taikoxyz/taiko-mono/pull/15053/files

Reply:
I'm not addressing the same thing, you are adding a new feature, I'm just optimizing the storage size for the old contract without a new feature.

Reply:
> TaikoL2EIP1559Configurable.sol

Done.

Reply:
> I'm not addressing the same thing, you are adding a new feature, I'm just optimizing the storage size for the old contract without a new feature.

Took over your size reductions here.